packages/dd-trace/test/telemetry/send-data.spec.js

```diff
@@ -12,40 +17,78 @@ describe('sendData', () => {
       '../exporters/common/request': request
     })
   })
+
   it('should call to request (TCP)', () => {
-    sendDataModule.sendData({ hostname: '', port: '12345', tags: { 'runtime-id': '123' } }, 'test', 'test', 'req-type')
+    sendDataModule.sendData({ hostname: '', port: '12345', tags: { 'runtime-id': '123' } },
+      application, 'test', 'req-type')
     expect(request).to.have.been.calledOnce
     const options = request.getCall(0).args[1]
 
     expect(options).to.deep.equal({
       method: 'POST',
       path: '/telemetry/proxy/api/v2/apmtelemetry',
       headers: {
-        'content-type': 'application/json',
-        'dd-telemetry-api-version': 'v1',
-        'dd-telemetry-request-type': 'req-type'
+        'Content-Type': 'application/json',
+        'DD-Telemetry-API-Version': 'v1',
+        'DD-Telemetry-Request-Type': 'req-type',
+        'DD-Client-Library-Language': application.language_name,
+        'DD-Client-Library-Version': application.tracer_version
       },
       url: undefined,
       hostname: '',
       port: '12345'
     })
   })
+
   it('should call to request (UDP)', () => {
-    sendDataModule.sendData({ url: 'unix:/foo/bar/baz', tags: { 'runtime-id': '123' } }, 'test', 'test', 'req-type')
+    sendDataModule.sendData({ url: 'unix:/foo/bar/baz', tags: { 'runtime-id': '123' } },
+      application, 'test', 'req-type')
     expect(request).to.have.been.calledOnce
     const options = request.getCall(0).args[1]
 
     expect(options).to.deep.equal({
       method: 'POST',
       path: '/telemetry/proxy/api/v2/apmtelemetry',
       headers: {
-        'content-type': 'application/json',
-        'dd-telemetry-api-version': 'v1',
-        'dd-telemetry-request-type': 'req-type'
+        'Content-Type': 'application/json',
+        'DD-Telemetry-API-Version': 'v1',
+        'DD-Telemetry-Request-Type': 'req-type',
+        'DD-Client-Library-Language': application.language_name,
+        'DD-Client-Library-Version': application.tracer_version
       },
       url: 'unix:/foo/bar/baz',
       hostname: undefined,
       port: undefined
     })
   })
+
+  it('should add debug header if DD_TELEMETRY_DEBUG is present', () => {
+    sendDataModule = proxyquire('../../src/telemetry/send-data', {
+      '../exporters/common/request': request
+    })
```

Comment:
isn't that already done in the beforeach ?

Reply:
true